### PR TITLE
Add extraction rules examples

### PR DESCRIPTION
**Pull request by navarone-feekery**
### Part of general docs improvements

- Rename `CONTENT_EXTRACTION.md` to `EXTRACTION_RULES.md` to better align with the actual feature name
- Add example configurations to `EXTRACTION_RULES.md`

### CODE REVIEW

**Review by seanstory — DISMISSED** [review]

File: docs/features/EXTRACTION_RULES.md [hunk 1]
```diff
@@ -113,3 +118,121 @@ Value can be anything except `null`.
...
+            selector: ".city"
+            join_as: "array"
+            source: "html"
+```
+
+In this example, the output document will include the following field on top of the standard crawl result fields:
+
+```json
+{
+  "cities": ["Neverwinter", "Waterdeep", "Menzoberranzan"]
```
Comment (marciw) [inline docs/features/EXTRACTION_RULES.md:165]:
```suggestion
  "cities": ["Summerstay", "Drenchwell", "Mezzoterran"]
```

Comment (marciw) [reply docs/features/EXTRACTION_RULES.md:165]: (or whatever you choose for lines 135-37)

File: docs/features/EXTRACTION_RULES.md [hunk 2]
```diff
@@ -113,3 +118,121 @@ Value can be anything except `null`.
...
+The HTML for this page looks like this:
+
+```HTML
+<!DOCTYPE html>
+<html>
+  <body>
+    <div>Cities:</div>
+    <div class="city">Neverwinter</div>
+    <div class="city">Waterdeep</div>
+    <div class="city">Menzoberranzan</div>
```
Comment (marciw) [inline docs/features/EXTRACTION_RULES.md:137]:
```suggestion
    <div class="city">Summerstay</div>
    <div class="city">Drenchwell</div>
    <div class="city">Mezzoterran</div>
```

Comment (marciw) [reply docs/features/EXTRACTION_RULES.md:137]: Just suggesting nonsense names instead of Forgotten Realms cities :)

Feel free to use something else!

**Review by marciw — APPROVED** [review]
Nice example additions! 🥇 Made some minor suggestions

**Review by jedrazb — APPROVED** [review]

File: docs/features/EXTRACTION_RULES.md (hunk 2 above)
Comment (seanstory) [reply docs/features/EXTRACTION_RULES.md]: 😢 I thought the pop references were cute

Comment (navarone-feekery) [reply docs/features/EXTRACTION_RULES.md]: I'll save them for unit tests :D

**Pull request merged**